grpc/src/main/java/com/linecorp/armeria/server/grpc/HttpJsonTranscodingService.java

```diff
@@ -416,6 +427,58 @@ private static Descriptor findTypeDescriptor(FileDescriptor file, FieldDescripto
         return file.findMessageTypeByName(messageType.getName());
     }
 
+    // to make it more efficient, we calculate whether extract response body one time
+    // if there is no matching toplevel field, we set it to null
+    @Nullable
+    private static String calculateResponseBody(List<FieldDescriptor> topLevelFields, String responseBody) {
+        if (StringUtil.isNullOrEmpty(responseBody)) {
+            return null;
+        }
+        for (FieldDescriptor fieldDescriptor: topLevelFields) {
+            if (fieldDescriptor.getName().equals(responseBody)) {
+                return responseBody;
+            }
+        }
+        return null;
+    }
+
+    @Nullable
+    private static Function<HttpData, HttpData> generateResponseBodyConverter(TranscodingSpec spec) {
+        @Nullable final String responseBody = spec.responseBody;
+        if (responseBody == null) {
+            return null;
+        } else {
+            return httpData -> {
+                final byte[] array = httpData.array();
+                try {
+                    final JsonNode jsonNode = JacksonUtil.readValue(array, JsonNode.class);
+                    // we try to convert lower snake case response body to camel case
+                    final String lowerCamelCaseResponseBody =
+                            CaseFormat.LOWER_UNDERSCORE.to(CaseFormat.LOWER_CAMEL, responseBody);
+                    final Iterator<Entry<String, JsonNode>> fields = jsonNode.fields();
+                    while (fields.hasNext()) {
+                        final Entry<String, JsonNode> entry = fields.next();
+                        final String fieldName = entry.getKey();
+                        final JsonNode responseBodyJsonNode = entry.getValue();
+                        // try to match field name and response body
+                        // 1. by default the marshaller would use lowerCamelCase in json field
+                        // 2. when the marshaller use original name in .proto file when serializing messages
+                        if (fieldName.equals(lowerCamelCaseResponseBody) ||
+                            fieldName.equals(responseBody)) {
+                            httpData.close();
+                            final byte[] bytes = JacksonUtil.writeValueAsBytes(responseBodyJsonNode);
```

Comment:
I didn't know that a `mapper` is defined already. Could we use it?
```suggestion
                            final byte[] bytes = mapper.writeValueAsBytes(responseBodyJsonNode);
```